File: app/src/main/java/com/sg/moviesindex/adapter/SearchAdapter.java
package com.sg.moviesindex.adapter;

import android.content.Context;
import android.content.Intent;
import android.database.Cursor;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.cursoradapter.widget.CursorAdapter;

import com.sg.moviesindex.R;
import com.sg.moviesindex.model.tmdb.Movie;
import com.sg.moviesindex.view.MoviesInfo;

import java.util.ArrayList;


public class SearchAdapter extends CursorAdapter {
  private final LayoutInflater mLayoutInflater;
  private final Context mContext;
  private final ArrayList<Movie> movies;

  public SearchAdapter(Context context, Cursor c, boolean autoRequery, ArrayList<Movie> movies) {
    super(context, c, autoRequery);
    mContext = context;
    mLayoutInflater = LayoutInflater.from(context);
    this.movies = movies;
  }

  @Override
  public View newView(Context context, Cursor cursor, ViewGroup parent) {
    return mLayoutInflater.inflate(R.layout.search_list, parent, false);
  }

  @Override
  public View getView(int position, View convertview, ViewGroup arg2) {
    if (convertview == null) {
      LayoutInflater inflater = LayoutInflater.from(mContext);
      convertview = inflater.inflate(R.layout.search_list,
          null);
    }
    convertview.setTag(position);
    return super.getView(position, convertview, arg2);
  }

  @Override
  public void bindView(final View view, Context context, final Cursor cursor) {
    int columnIndex = cursor.getColumnIndex("text");
    if (columnIndex != -1) {
      String title = cursor.getString(columnIndex);
      TextView textView = view.findViewById(R.id.textView2);
      textView.setText(title);
      view.setOnClickListener(v -> {
        int id = (Integer) view.getTag();//here is the position
        Movie movie = movies.get(id);
        Intent i = new Intent(mContext, MoviesInfo.class);
        i.putExtra("movie", movie);
        mContext.startActivity(i);
      });
    }

  }
}

File: app/src/main/java/com/sg/moviesindex/ui/search/SearchAdapter.kt
package com.sg.moviesindex.ui.search

import android.content.Context
import android.content.Intent
import android.database.Cursor
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.cursoradapter.widget.CursorAdapter
import com.sg.moviesindex.R
import com.sg.moviesindex.data.local.Movie
import com.sg.moviesindex.ui.details.MovieDetailActivity

class SearchAdapter(
  private val context: Context,
  c: Cursor?,
  autoRequery: Boolean,
  private val movies: ArrayList<Movie>,
) : CursorAdapter(context, c, autoRequery) {
  override fun newView(
    context: Context,
    cursor: Cursor,
    parent: ViewGroup,
  ): View = LayoutInflater.from(context).inflate(R.layout.search_list, parent, false)

  override fun bindView(
    view: View,
    context: Context,
    cursor: Cursor,
  ) {
    val textView = view.findViewById<TextView>(R.id.textView2)
    val text = cursor.getString(cursor.getColumnIndexOrThrow("text"))
    textView.text = text
  }

  override fun getView(
    position: Int,
    convertView: View?,
    parent: ViewGroup,
  ): View {
    val view = super.getView(position, convertView, parent)
    view.setOnClickListener {
      if (position < movies.size) {
        val movie = movies[position]
        val intent =
          Intent(context, MovieDetailActivity::class.java).apply {
            putExtra("movie", movie)
          }
        context.startActivity(intent)
      }
    }
    return view
  }
}
